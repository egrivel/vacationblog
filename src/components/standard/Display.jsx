import React from 'react';
import PropTypes from 'prop-types';

import Formrow from './Formrow.jsx';

class Display extends React.Component {
  static propTypes = {
    fieldId: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.string
  };

  static defaultProps = {
    value: ''
  }

  render() {
    return (
      <Formrow
        key={'k-' + this.props.fieldId}
        label={this.props.label}
      >
        <span>
          {this.props.value}
        </span>
      </Formrow>
    );
  }
}

export default Display;
