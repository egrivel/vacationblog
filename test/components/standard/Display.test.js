import React from 'react';
import renderer from 'react-test-renderer';

import Display from '../../../src/components/standard/Display';

/*
 * Test strategy
 * The component has one optional attribute (the value) and no logic.
 * Confirm that the component renders correctly both with and without
 * the attribute.
 */
describe('components/standard/Display', () => {
  const testFieldId = 'test-field-id';
  const testLabel = 'Test Label';
  const testValue = 'Test Value';

  describe('#render', () => {
    test('renders with label and value', () => {
      const render = renderer.create(
        <Display
          fieldId={testFieldId}
          label={testLabel}
          value={testValue}
        />
      );
      expect(render.toJSON()).toMatchSnapshot();
    });

    test('renders with label but no value', () => {
      const render = renderer.create(
        <Display
          fieldId={testFieldId}
          label={testLabel}
        />
      );
      expect(render.toJSON()).toMatchSnapshot();
    });
  });
});
