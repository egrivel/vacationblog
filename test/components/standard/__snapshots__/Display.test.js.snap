// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/standard/Display renders with label and value 1`] = `
<div>
  <div
    className="formLabel"
  >
    Test Label
  </div>
  <div
    className="formValue"
  >
    <span>
      Test Value
    </span>
  </div>
</div>
`;

exports[`components/standard/Display renders with label but no value 1`] = `
<div>
  <div
    className="formLabel"
  >
    Test Label
  </div>
  <div
    className="formValue"
  >
    <span>
      
    </span>
  </div>
</div>
`;
